nfcauth.c

```diff
@@ -65,21 +66,24 @@ nfcauth_get_targets (char **targets[])
         .nbr = NBR_UNDEFINED
     };
 
+    //printf("%d devices\n",device_count);
     for (i = 0; i < device_count; i++) {
         nfc_device* initiator = nfc_open (context, devices[i]);
         if (initiator) {
             nfc_initiator_init (initiator);
             nfc_target ant[MAX_TARGET];
             int res;
+            //printf("searching targets...");
             if ((res = nfc_initiator_list_passive_targets (initiator, nm, ant, MAX_TARGET)) >= 0) {
+            //printf("found %d targets\n",res);
                 size_t  iTarget;
                 target_count = res;
                 for (iTarget = 0; iTarget < target_count; iTarget++) {
-                    if (((*targets)[ret] = malloc (2 * ant[iTarget].nti.nai.szUidLen + 1))) {
+                    if (((*targets)[ret] = malloc (2 * ant[iTarget].nti.nai.szAtsLen + 1))) {
```

Comment:
this sounds like a very bad idea.
ATS is not unique at all, anybody with the same phone model will be able to unlock your machine, is it what you want?

Reply:
Hi,
thanks for your input. I do of course know that this is not the best way to go about things - maybe it is not far from actually being one of the worst. But this is something I play with while trying to figure out how to write an actual smart card app for android. It illustrates certain topics  I discuss in an ongoing series of articles on my home page (in german). So without that context it can come over as terrible - I do realize that now.
Maybe I should have stressed the fact that this is only something to play with and not meant to
actually use in a "production environment" - i will state so in the README I am going to update.
For a little context - my actual goal is to implement an idea from this master thesis i found:
http://sar.informatik.hu-berlin.de/research/publications/SAR-PR-2014-08/SAR-PR-2014-08_.pdf.
It even has its own repository: https://github.com/eriknellessen/Virtual-Keycard

Best regards,

Juergen Key

Reply:
ok, I got a notification on your commit and reacted on it because it's part of a Pull Request. I'll cancel this pull request as visibly it's not meant to be merged in the main repo.